pkg/collector/corechecks/network-devices/cisco-sdwan/sdwan.go

```diff
@@ -0,0 +1,211 @@
+	deviceTags := payload.GetDevicesTags(c.config.Namespace, devices)
+	uptimes := payload.GetDevicesUptime(devices)
+
+	interfaces := payload.ConvertInterfaces(vEdgeInterfaces, cEdgeInterfaces)
+	interfacesMetadata, interfacesMap := payload.GetInterfacesMetadata(c.config.Namespace, interfaces)
+	ipAddressesMetadata := payload.GetIPAddressesMetadata(c.config.Namespace, interfaces)
+
+	c.metricsSender.SetDeviceTags(deviceTags)
+
+	if !c.config.NoMetadata {
+		c.metricsSender.SendMetadata(devicesMetadata, interfacesMetadata, ipAddressesMetadata)
+	}
+	c.metricsSender.SendDeviceMetrics(deviceStats)
+	c.metricsSender.SendInterfaceMetrics(interfaceStats, interfacesMap)
+	c.metricsSender.SendUptimeMetrics(uptimes)
+	c.metricsSender.SendAppRouteMetrics(appRouteStats)
+	c.metricsSender.SendControlConnectionMetrics(controlConnectionsState)
+	c.metricsSender.SendOMPPeerMetrics(ompPeersState)
+	c.metricsSender.SendBFDSessionMetrics(bfdSessionsState)
+	c.metricsSender.SendDeviceCountersMetrics(deviceCounters)
+
+	// Commit
+	c.metricsSender.Commit()
+	log.Info("Done running Cisco SD-WAN check")
+
+	return nil
+}
+
+// Configure the Cisco SD-WAN check
+func (c *CiscoSdwanCheck) Configure(senderManager sender.SenderManager, integrationConfigDigest uint64, rawInstance integration.Data, rawInitConfig integration.Data, source string) error {
+	// Must be called before c.CommonConfigure
+	c.BuildID(integrationConfigDigest, rawInstance, rawInitConfig)
+
+	err := c.CommonConfigure(senderManager, integrationConfigDigest, rawInitConfig, rawInstance, source)
+	if err != nil {
+		return err
+	}
+
+	var instanceConfig checkCfg
+	err = yaml.Unmarshal(rawInstance, &instanceConfig)
+	if err != nil {
+		return err
+	}
+	c.config = instanceConfig
+
+	if c.config.Namespace == "" {
+		c.config.Namespace = "cisco-sdwan"
+	} else {
+		namespace, err := utils.NormalizeNamespace(c.config.Namespace)
+		if err != nil {
+			return err
+		}
+		c.config.Namespace = namespace
+	}
+
+	var clientOptions []client.ClientOptions
+
+	if instanceConfig.Insecure || instanceConfig.CAFile != "" {
+		options, err := client.WithTLSConfig(instanceConfig.Insecure, instanceConfig.CAFile)
+		if err != nil {
+			return err
+		}
+
+		clientOptions = append(clientOptions, options)
+	}
+
+	if instanceConfig.MaxAttempts > 0 {
+		clientOptions = append(clientOptions, client.WithMaxAttempts(instanceConfig.MaxAttempts))
+	}
+
+	if instanceConfig.MaxPages > 0 {
+		clientOptions = append(clientOptions, client.WithMaxPages(instanceConfig.MaxPages))
+	}
+
+	if instanceConfig.MaxCount > 0 {
+		clientOptions = append(clientOptions, client.WithMaxCount(instanceConfig.MaxCount))
+	}
+
+	if instanceConfig.Lookback > 0 {
+		clientOptions = append(clientOptions, client.WithLookback(time.Second*time.Duration(instanceConfig.Lookback)))
+	}
+
+	// Create Cisco SD-WAN API client
+	c.client, err = client.NewClient(instanceConfig.Hostname, instanceConfig.Username, instanceConfig.Password, instanceConfig.UseHTTP, clientOptions...)
```

Comment:
Should we reset the client on every check run? This way in case the client goes unhealthy (for instance authentication starts to fail because of bad cookies or whatever), the agent can recover from it.
I'm usually worried of keeping too much state because of that state goes bad the only way to recover is to restart the Agent which would be a poor experience

Reply:
Good point! Addressed this in 8d93cb6